lego/apps/surveys/serializers.py

```diff
@@ -0,0 +1,144 @@
+from rest_framework import exceptions, serializers
+
+from lego.apps.events.serializers.events import EventReadSerializer
+from lego.apps.surveys.constants import QUESTION_TYPES
+from lego.apps.surveys.models import Answer, Option, Question, Submission, Survey
+# from lego.apps.users.serializers.users import PublicUserSerializer
+from lego.utils.serializers import BasisModelSerializer
+
+
+class OptionSerializer(BasisModelSerializer):
+
+    class Meta:
+        model = Option
+        fields = ('id', 'option_text')
+
+
+class QuestionSerializer(BasisModelSerializer):
+    question_type = serializers.ChoiceField(choices=QUESTION_TYPES)
+    options = OptionSerializer(many=True, required=False, allow_null=True)
+
+    class Meta:
+        model = Question
+        fields = ('id', 'question_type', 'question_text', 'mandatory', 'options')
+
+
+class AnswerSerializer(BasisModelSerializer):
+    question = QuestionSerializer()
+    selected_options = OptionSerializer(many=True, required=False)
+
+    class Meta:
+        model = Answer
+        fields = ('id', 'submission', 'question', 'answer_text', 'selected_options')
+
+
+class AnswerCreateAndUpdateSerializer(BasisModelSerializer):
+
+    class Meta:
+        model = Answer
+        fields = ('id', 'question', 'answer_text', 'selected_options')
+
+
+class SurveyReadSerializer(BasisModelSerializer):
+    event = EventReadSerializer()
+
+    class Meta:
+        model = Survey
+        fields = ('id', 'title', 'active_from', 'event')
+
+
+class SubmissionReadSerializer(BasisModelSerializer):
+    # Do we want user and survey as nested objects?
+    # user = PublicUserSerializer()
+    # survey = SurveyReadSerializer()
+    answers = AnswerSerializer(many=True)
```

Comment:
Do we need answers in the list view?

Reply:
Remember that this is the Submission list view, not the Survey list view. This is the endpoint that will be called when the user click a specific survey and navigates to "Answers". I suppose this view should actually just return some aggregated data, such as number of submissions and how many users selected each option for each question.